content/blog/using-apache-spark-dataframes-for-processing-of-tabular-data-1.md

```diff
@@ -0,0 +1,215 @@
+
+Finally, the RDD action saveAsTextFile(path) writes the elements of the dataset as a text file (or set of text files) in the outputFile directory.
+
+```Java
+        String outputFile = args[1];
+        // Save the word count back out to a text file, causing evaluation.
+        counts.saveAsTextFile(outputFile);                
+```
+
+**Example Word Count App in Scala**
+
+Here is the same example in Scala:
+
+![](/img/tutorial_spark8.png)
+
+**Building a Simple Application**
+
+Spark can be linked into applications in either Java, Scala, or Python.\
+Maven is a popular package management tool for Java-based languages that lets you link to libraries in public repositories. In Java and Scala, you give your application a Maven dependency on the spark-core artifact. The current Spark version is 1.6.1 and the Maven coordinates are:
+
+```xml
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-core_2.10</artifactId>
+            <version>1.6.1</version>
+            <scope>provided</scope>
+        </dependency>
+```
+
+You can build with Maven using IDEs like Eclipse or NetBeans, and then copy the JAR file to your MapR Sandbox, or you can install Maven on your sandbox and build from the Linux command line.
+
+![](/img/tutorial_spark9.png)
+
+You can use scp to copy your JAR file to the MapR Sandbox:
+
+![](/img/tutorial_spark10.png)
+
+* See below for an example of using scp from the command line:\
+  use userid: user01 and password: mapr.
+
+  For VMWare use: `$ scp nameoffile.jar user01@ipaddress:/user/user01/.`
+
+  For Virtualbox use: `$ scp -P 2222 nameoffile.jar user01@127.0.0.1:/user/user01/.`
+
+**Running Your Application**
+
+First find the version of Spark on the sandbox with `ls /opt/mapr/spark/` Then you can use the spark commands in the /opt/mapr/spark/spark-version/bin directory.
+
+You use the `bin/spark-submit` script to launch your application. This script takes care of setting up the classpath with Spark and its dependencies. Here is the spark-submit format:
+
+```xml
+./bin/spark-submit \
+  --class <main-class>
+  --master <master-url> \
+  <application-jar> \
+  [application-arguments]
+```
+
+Here is the spark-submit command for our example, passing the input file and output directory as arguments to the main method of the `JavaWordCount` class:
+
+```scala
+/opt/mapr/spark/spark-1.6.1/bin/spark-submit --class example.wordcount.JavaWordCount --master yarn \
+  sparkwordcount-1.0.jar /user/user01/input/alice.txt /user/user01/output
+```
+
+Here is the spark-submit command to run the `scala SparkWordCount`:
+
+```scala
+/opt/mapr/spark/spark-1.6.1/bin/spark-submit --class SparkWordCount --master yarn \
+  sparkwordcount-1.0.jar /user/user01/input/alice.txt /user/user01/output
+```
+
+This concludes the Getting Started with Spark on the MapR Sandbox tutorial. You can download the example Maven project code here: <https://github.com/caroljmcdonald/sparkwordcountapp>
```

Comment:
Suggest to hotlink the URL. You can download the example Maven project code [here](https://github.com/caroljmcdonald/sparkwordcountapp).